packages/server/src/api/controllers/view/index.ts

```diff
@@ -48,18 +49,16 @@ export async function save(ctx: Ctx) {
   if (!view.meta.schema) {
     view.meta.schema = table.schema
   }
-  table.views[viewName] = view.meta
+  table.views[viewName] = { ...view.meta, name: viewName }
```

Comment:
There was some weird inconsistencies with where the name of a view was stored, especially in the frontend code that calls this endpoint. I've rewritten most of that, and now I'm also just ensuring that the view name is properly saved the view itself.